uaclient/tests/test_apt.py

```diff
@@ -347,140 +347,190 @@ def test_errors_on_apt_helper_process_timeout(
-        """
-        )
+        if series == "xenial":
+            expected_content = dedent(
+                """\
+                deb http://fakerepo xenial-one main
+                # deb-src http://fakerepo xenial-one main
+                deb http://fakerepo xenial-updates main
+                # deb-src http://fakerepo xenial-updates main
+            """
+            )
+        else:
+            expected_content = dedent(
+                """\
+                # Written by ubuntu-advantage-tools
+
+                Types: deb
+                URIs: http://fakerepo
+                Suites: noble-one noble-updates
+                Components: main
+                Signed-By: /usr/share/keyrings/keyring
+            """
+            )
+
         assert expected_content == system.load_file(repo_file)
 
+    @pytest.mark.parametrize("series", ("xenial", "noble"))
     @mock.patch("uaclient.apt.gpg.export_gpg_key")
     @mock.patch("uaclient.system.subp")
     @mock.patch("uaclient.apt.get_apt_auth_file_from_apt_config")
     @mock.patch("uaclient.apt.assert_valid_apt_credentials")
-    @mock.patch(
-        "uaclient.system.get_release_info",
-        return_value=mock.MagicMock(series="xenial"),
-    )
+    @mock.patch("uaclient.system.get_release_info")
     def test_add_auth_apt_repo_comments_updates_suites_on_non_update_machine(
         self,
         m_get_release_info,
         m_valid_creds,
         m_get_apt_auth_file,
         m_subp,
         m_gpg_export,
+        series,
         tmpdir,
     ):
         """Skip any apt suites that don't match the current series."""
         repo_file = tmpdir.join("repo.conf").strpath
         auth_file = tmpdir.join("auth.conf").strpath
         m_get_apt_auth_file.return_value = auth_file
-        # apt policy without xenial-updates enabled
-        origin = "test-origin"
+        # apt policy without series-updates enabled
         m_subp.return_value = (
-            POST_INSTALL_APT_CACHE_NO_UPDATES.format("xenial", origin),
+            POST_INSTALL_APT_CACHE_NO_UPDATES.format(series, "test-origin"),
             "",
         )
+        m_get_release_info.return_value = mock.MagicMock(series=series)
 
         add_auth_apt_repo(
             repo_filename=repo_file,
             repo_url="http://fakerepo",
             credentials="mycreds",
-            suites=("xenial-one", "xenial-updates", "trusty-gone"),
+            suites=(
+                "{}-one".format(series),
+                "{}-updates".format(series),
+                "trusty-gone",
+            ),
             keyring_file="keyring",
         )
 
-        expected_content = dedent(
-            """\
-            deb http://fakerepo xenial-one main
-            # deb-src http://fakerepo xenial-one main
-            # deb http://fakerepo xenial-updates main
-            # deb-src http://fakerepo xenial-updates main
-        """
-        )
+        if series == "xenial":
```

Comment:
Also best to use a check using `if series in apt.SERIES_NOT_USING_DEB822` perhaps?